sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_receiver_async.py

```diff
@@ -826,6 +828,78 @@ async def peek_messages(
         ):
             return messages
 
+    async def delete_messages(
+        self,
+        *,
+        max_message_count: Optional[int] = None,
+        before_enqueued_time_utc: Optional[datetime.datetime] = None,
+    ) -> int:
+        """  This operation deletes messages in the queue that are older than the specified enqueued time,
+         up to 4,000 messages at a time.
+
+        :param int or None max_message_count: The maximum number of messages to delete. The default value is 1.
+        :param datetime.datetime or None before_enqueued_time_utc: The UTC datetime value before which all messages
+         should be deleted.The default value is None, meaning all messages in the queue will be considered.
+        :rtype: int
+
+        """
+        self._check_live()
+        await self._open()
+
+        message_count = max_message_count if max_message_count else 4000
+
+        message = {
+            MGMT_REQUEST_ENQUEUED_TIME_UTC: before_enqueued_time_utc if before_enqueued_time_utc
+                else datetime.datetime.now(datetime.timezone.utc),
+            MGMT_REQUEST_MAX_MESSAGE_COUNT: message_count,
+        }
+
+        self._populate_message_properties(message)
+        handler = functools.partial(mgmt_handlers.batch_delete_op, receiver=self, amqp_transport=self._amqp_transport)
+        deleted = await self._mgmt_request_response_with_retry(
+            REQUEST_RESPONSE_DELETE_BATCH_OPERATION, message, handler
+        )
+
+        links = get_receive_links(deleted)
+        with receive_trace_context_manager(self, span_name=SPAN_NAME_PEEK, links=links):
+            return deleted
+        
+    async def purge_messages(
+        self,
+        *,
+        before_enqueued_time_utc: Optional[datetime.datetime] = None,
```

Comment:
would we need this kwarg here ?

Reply:
to allow users to purge at any point in time